src/data_access/SearchDataAccessObject.java

```diff
@@ -0,0 +1,78 @@
+package data_access;
+
+import entity.SearchChatMessage;
+import entity.SearchRequest;
+import okhttp3.*;
+import use_case.search.SearchDataAccessInterface;
+
+public class SearchDataAccessObject implements SearchDataAccessInterface {
+
+  @Override
+  public void getData(SearchRequest searchRequest) {
+
+    // Replace these placeholders with your actual Elasticsearch URL and API key
+    String esUrl = "";
+    String apiKey = "";
+
+    OkHttpClient client = new OkHttpClient();
+
+    // Define the Elasticsearch index to query
+    String index = "search-chats";
+
+    // Define your Elasticsearch query (in this case, a match query for the "name" field)
+    String query = "{ \"query\": { \"match\": { \"name\": \"1984\" } } }";
+
+    Request request =
+        new Request.Builder()
+            .url(esUrl + "/" + index + "/_search")
+            .addHeader("Authorization", "ApiKey " + apiKey)
+            .addHeader("Content-Type", "application/json")
+            .post(RequestBody.create(query, MediaType.get("application/json")))
+            .build();
+
+    try (Response response = client.newCall(request).execute()) {
+      if (response.isSuccessful()) {
+        searchRequest.changeQueryResponse(response.body().string());
+      } else {
+        searchRequest.changeQueryResponse("sorry, didn't find anything!");
+      }
+    } catch (Exception e) {
+      searchRequest.changeQueryResponse("sorry, didn't find anything!");
+    }
+  }
+
+  @Override
+  public void saveData(SearchChatMessage message) {
+    // Replace these placeholders with your actual Elasticsearch URL and API key
+    String esUrl = "your_elasticsearch_url";
```

Comment:
Please extract esUrl and apiKey into https://github.com/uoftm/csc207project/blob/main/src/data_access/Constants.java

Reply:
Ok, done